components/server/src/user/user-controller.ts

```diff
@@ -252,6 +254,10 @@ export class UserController {
             // clear cookies
             this.sessionHandlerProvider.clearSessionCookie(res, this.config);
 
+            // terminate all running workspaces
+            const user = req.user as User;
+            await this.userDeletionService.stopWorkspaces(user);
```

Comment:
Out of curiosity, is there no implementation of `stopWorkspace` on the WorkspaceService? Structurally, this feels odd to have the /logout endpoint depend on the user deletion service when it has nothing to do with deletion.

Reply:
Good point, I've removed the implementation in UserDeletionService and use the one in `WorkspaceStarter` (which we could rename to `WorkspaceService`).